sdks/python/apache_beam/runners/portability/fn_api_runner_test.py

```diff
@@ -188,22 +188,21 @@ def test_pardo_windowed_side_inputs(self):
               (9, list(range(7, 10)))]),
           label='windowed')
 
-  def test_flattened_side_input(self, extended=True):
+  def test_flattened_side_input(self):
     with self.create_pipeline() as p:
       main = p | 'main' >> beam.Create([None])
       side1 = p | 'side1' >> beam.Create([('a', 1)])
       side2 = p | 'side2' >> beam.Create([('b', 2)])
-      side3 = p | 'side3' >> beam.Create(['another type'])
```

Comment:
The fact that this was another type is intentional (specifically, we should handle this, possibly via transcoding).

Reply:
Figured it was intentional, was planning on letting you review this. As a general rule, flattening only works for matching types and Flink does not support flattening of different types. `side1`/`side2` use a KV list coder, whereas `side3` used a string list coder which lead to different result types.

Transcoding can be implemented, but it requires rules for translating different coders to a single result type. I think it would be worth adding a dedicated test for testing flattening rules for non-matching coder types.

Reply:
The original bug only manifested when transcoding was required in this particular context (flattening with different types alone didn't reproduce it). I would be OK with spiting this out into a separate test.

Reply:
Ok makes sense. If it required that particular context, I'll re-add the extension to the test.

Reply:
Are there transcoding rules specified somewhere else? The cases where it is applicable should be clear to users and runner authors. Would a separate test specifically for transcoding, even when it repeats the context partially, make sense?

Reply:
A runner can choose to do whatever it wants to implement this; the logical choice is to transcode everything to the coder specified on the output PCollection. 

Yes, I think a separate test would make sense.